fission/src/systems/physics/PhysicsSystem.ts

```diff
@@ -90,6 +100,64 @@ class PhysicsSystem extends WorldSystem {
         )
         ground.SetFriction(FLOOR_FRICTION)
         this._joltBodyInterface.AddBody(ground.GetID(), JOLT.EActivation_Activate)
+
+        this._bodyAssociations = new Map()
+    }
+
+    /**
+     * Get association to a given Jolt Body.
+     *
+     * @param bodyId BodyID to check for association
+     * @returns Association for given Body
+     */
+    public GetBodyAssociation<T extends object & BodyAssociated>(bodyId: Jolt.BodyID): T | undefined {
+        const res = this._bodyAssociations.get(bodyId.GetIndexAndSequenceNumber())
+        if (res) {
+            // Avoids error, simply returns undefined if invalid
+            return res as unknown as T
```

Comment:
I'm fairly certain that this cast doesn't actually properly type check and I don't know how to fix it

Reply:
I originally had a separate BodyAssociated object for gamepieces, but literally not matter what I did, it would somehow successfully force cast a RigidNodeAssociate to GamePieceAssociate. I had required object types that weren't allowed to be null just fully forced to be null and I have no idea why. You can do normal `as` but then it throws an error on fail instead of just returning undefined.

Reply:
Ok, maybe make a note about what kind of types T work, if you have an idea that might be helpful to anyone else using this function.

Reply:
Honestly the issue is with horrid generics support for TypeScript. I think I'm going to dispense with it and leave it to the user.

Reply:
Ok ya that sounds good. Generic traits in TS when?